### PR TITLE
Priority of PrometheusMeterRegistry; tests

### PR DESCRIPTION
**Pull request by ebullient**
Demonstrates and tests issue reported in #15219 

Additional fixes required in quarkiverse to set `@AlternativePriority` appropriately for other MeterRegistries

### CODE REVIEW

File: extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/export/PrometheusMeterRegistryProvider.java [hunk 1]
```diff
@@ -58,7 +58,7 @@ public CollectorRegistry collectorRegistry() {
 
     @Produces
     @Singleton
-    @AlternativePriority(Interceptor.Priority.APPLICATION + 100)
+    @AlternativePriority(Interceptor.Priority.APPLICATION)
```
Comment (gsmet) [inline extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/export/PrometheusMeterRegistryProvider.java:61]: Is it the correct priority for a framework bean?

Also, I wonder if we should make it work another way:
- use `@DefaultBean`
- have a qualifier with a name to define others
?

The current setup looks very weird to me. @mkouba could you have a look at this and the discussion here: https://github.com/quarkusio/quarkus/pull/15219 ?

**Review by gsmet — CHANGES_REQUESTED** [review]
Let's wait for @mkouba 's feedback.

File: extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/export/PrometheusMeterRegistryProvider.java (hunk 1 above)
Comment (ebullient) [reply extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/export/PrometheusMeterRegistryProvider.java:61]: There is a valid use case to use both the bean automatically configured by the runtime AND an additional, custom bean configured manually, which is why I did not use a DefaultBean to begin with.
 
I am trying to make a value that is more obvious to supersede, if desired (though I don't think most people will(

Comment (ebullient) [reply extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/export/PrometheusMeterRegistryProvider.java:61]: I need to think on it a bit more, but using DefaultBean and auto-ejecting the framework defined bean, while perhaps more obvious on the surface, isn't the right answer.. but the @AlternativePriority isn't consistently selecting/omitting the way I expected.

**Comment by ebullient** [comment]
See #17481 instead

**Pull request closed without merging**